### PR TITLE
Upgrade Python to 3.10

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/959

Upgrade Python from 3.9 to 3.10

Notes:
- using a branch of utils for now
- need to switch from the library `orderedset` to `ordered_set` since the former does not support 3.10 and is possibly abandonware.

# Test instructions | Instructions pour tester la modification

See if admin still works!

### CODE REVIEW

File: requirements-app.txt [hunk 1]
```diff
@@ -44,6 +44,7 @@ awscli-cwlogs>=1.4.6,<1.5
 # Putting upgrade on hold due to v1.0.0 using sha512 instead of sha1 by default
 itsdangerous==2.0.1  # pyup: <1.0.0
 
-git+https://github.com/cds-snc/notifier-utils.git@49.1.0#egg=notifications-utils
+# git+https://github.com/cds-snc/notifier-utils.git@49.1.0#egg=notifications-utils
+git+https://github.com/cds-snc/notifier-utils.git@python-3.10#egg=notifications-utils
```
Comment (sastels) [inline requirements-app.txt:48]: change when utils gets merged

File: requirements.txt [hunk 2]
```diff
@@ -46,7 +46,8 @@ awscli-cwlogs>=1.4.6,<1.5
 # Putting upgrade on hold due to v1.0.0 using sha512 instead of sha1 by default
 itsdangerous==2.0.1  # pyup: <1.0.0
 
-git+https://github.com/cds-snc/notifier-utils.git@49.1.0#egg=notifications-utils
+# git+https://github.com/cds-snc/notifier-utils.git@49.1.0#egg=notifications-utils
+git+https://github.com/cds-snc/notifier-utils.git@python-3.10#egg=notifications-utils
```
Comment (sastels) [inline requirements.txt:50]: change when utils gets merged

File: application.py [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
-    release="notify-admin@" + os.environ.get("GIT_SHA", ""),
-)
+# sentry_sdk.init(
+#     dsn=os.environ.get("SENTRY_URL", ""),
+#     integrations=[
+#         FlaskIntegration(),
+#         RedisIntegration(),
+#     ],
+#     release="notify-admin@" + os.environ.get("GIT_SHA", ""),
+# )
```
Comment (sastels) [inline application.py:24]: Sentry causes failures due to our old eventlet. We are not using sentry in production. Will discuss removing offline.

File: requirements.txt [hunk 4]
```diff
@@ -74,9 +75,9 @@ MarkupSafe==2.0.1
 marshmallow==3.18.0
 mistune==0.8.4
 openpyxl==3.0.10
-orderedset==2.0.3
+ordered-set==4.1.0
```
Comment (sastels) [inline requirements.txt:78]: orderedset doesn't work with 3.10 and seems abandoned, people have switched to ordered-set.

**Review by smcmurtry — DISMISSED** [review]
I re-tested everything I mentioned in [this comment](https://github.com/cds-snc/notification-utils/pull/154#pullrequestreview-1196618013) plus I tested changing some things in settings. I didn't find any bugs.

Looks good to me 👍

**Comment by sastels** [comment]
😞 The gunicorn / eventlet setup we run in the k8s docker container doesn't work with 3.10
https://github.com/eventlet/eventlet/issues/687
Verified by manually building and running `ci/Dockerfile`.
Going to see if I can upgrade these a bit and get things working...

**Comment by sastels** [comment]
So having gunicorn use gevent I can run the `ci/Dockerfile` image locally (hooked up to a devcontainer api) and send notifications. Next step is to merge into staging and do some perf tests there to ensure that we have similar performance as with eventlet.

**Review by smcmurtry — APPROVED** [review]
Looks good, but let's merge this after we fix CI in [this PR](https://github.com/cds-snc/notification-admin/pull/1408) incase we need to revert this

**Pull request merged**